plugins/output/adapt/importsource.js

```diff
@@ -203,15 +184,38 @@ function ImportSource(req, done) {
             repository: repository,
             createdBy: app.usermanager.getCurrentUser()._id
           };
-          if(!fileMeta) {
-            return doneAsset(new helpers.ImportError('No metadata found for asset: ' + assetName));
-          }
-          helpers.importAsset(fileMeta, metadata, doneAsset);
+
+          if (!assetJson) helpers.importAsset(fileMeta, metadata, doneAsset);
```

Comment:
```suggestion
          if (!assetJson) {
            helpers.importAsset(fileMeta, metadata, doneAsset);
            return;
          }
```

Reply:
I have just tested this with courses that do and don't have an assets.json file, this works for me.

@tomgreenfield was correct, the original lack of return here caused problems if an asset.json wasn't present